src/Lumi/Components2/Button.purs

```diff
@@ -1,88 +1,552 @@
+                }
+            )
+
+-- | `loadingContent` sets the content to display while a link button
+-- | is in its loading state. The size of the link button will always
+-- | fit the larger content, regardless which state it's in.
+loadingContent :: Array JSX -> forall r. PropsModifier ( component :: LinkButton, loadingContent :: Maybe (Array JSX) | r )
+loadingContent a =
+  propsModifier _ { loadingContent = Just a }
+
+recolor :: forall c. (ColorMap Color -> Color) -> ButtonModifier c
+recolor f =
+  style
+    ( \theme@(LumiTheme { colors: colors@{ black, white } }) ->
+        let
+          shades =
+            shade { hue: f colors, black, white }
+        in
+          css
+            { "--button-hue": color shades.hue
+            , "--button-hue-darker": color shades.hueDarker
+            , "--button-hue-darkest": color shades.hueDarkest
+            , "--button-hue-disabled": color shades.hueDisabled
+            , "--button-grey1": color shades.grey1
+            , "--button-grey2": color shades.grey2
+            , "--button-black": color shades.black
+            , "--button-white": color shades.white
+            }
+    )
+
+varButtonHue :: StyleProperty
+varButtonHue = var "--button-hue"
+
+varButtonHueDarker :: StyleProperty
+varButtonHueDarker = var "--button-hue-darker"
+
+varButtonHueDarkest :: StyleProperty
+varButtonHueDarkest = var "--button-hue-darkest"
+
+varButtonHueDisabled :: StyleProperty
+varButtonHueDisabled = var "--button-hue-disabled"
+
+varButtonGrey1 :: StyleProperty
+varButtonGrey1 = var "--button-grey1"
+
+varButtonGrey2 :: StyleProperty
+varButtonGrey2 = var "--button-grey2"
+
+varButtonBlack :: StyleProperty
+varButtonBlack = var "--button-black"
+
+varButtonWhite :: StyleProperty
+varButtonWhite = var "--button-white"
+
+-- | A form submit button. This helper takes the button state
+-- | as an argument because a form's buttons are generally
+-- | tied to the validity and `onSubmit` behavior of the form,
+-- | rather than providing an `onPress` action to the button
+-- | itself.
+submit :: forall c. ButtonState -> ButtonModifier c
+submit state = propsModifier _ { type = Submit, state = state }
+
+-- | A form reset button. This helper takes the button state
+-- | as an argument because a form's buttons are generally
+-- | tied to the validity and `onSubmit` behavior of the form,
+-- | rather than providing an `onPress` action to the button
+-- | itself.
+reset :: forall c. ButtonState -> ButtonModifier c
+reset state = propsModifier _ { type = Reset, state = state }
+
+-- | A button with customized `onPress` behavior. The button will
+-- | automatically display a loading state while the action is in-progress.
+-- | Using `onPress` on a button multiple times chains the effects together
+-- | from the first applied, out.
+onPress :: forall c. Aff Unit -> ButtonModifier c
+onPress a = propsModifier \props -> props { onPress = props.onPress *> a }
```

Comment:
This might not be necessary but it seemed like it could be useful to build a button with some base behavior and pass it to something else to add its own behavior, a little bit like DOM event bubbling. This seems like the right direction to me but I'm curious what anyone else things.

Reply:
I love this! Maybe if you want we could also expose a more flexible variant like this?
```purescript
onPress' :: forall c. (Aff Unit -> Aff Unit) -> ButtonModifier c
onPress' f = propsModifier \props -> props { onPress = f props.onPress }
```

Not sure if it would be useful though.